### PR TITLE
[IGNORE] Update UI build dependencies, require Node 18, fix Storybook CI

### PR DESCRIPTION
**Pull request by sjcobb**
Attempting to revive branch Gladorme:ui/deps-update in PR #1207

Testing whether build is green and CI actions pass w/o the weird `ci / build-storybook` issues from before

## TODO
- [x] Fix Storybook tests in CI using `npx wait-on tcp:127.0.0.1:6006`
- [x] Remove numbro from components package (those utils moved to core)
- [x] Regen failing snapshots and confirm diffs are expected 
- [ ] More manual testing
  - _I quickly tested locally, but if someone else has time to confirm the app is working as expected in areas we don't have e2e coverage_

## Notes
- nvm install v18.16.0
- nvm use
- node -v
  - v18.16.0

### CODE REVIEW

**Comment by sjcobb** [comment]
Ugh build-storybook is taking 33 minutes and counting so far, not a good sign, maybe we need to upgrade Storybook as part of this PR 🤔

**Comment by sjcobb** [comment]
Alright found a related issue with a fix! See: https://github.com/storybookjs/test-runner/issues/301

“this can be resolved by updating the npx wait-on command to `npx wait-on tcp:127.0.0.1:6006` (or whatever the correct address is for the http-server”

I don’t have time to dig further into that to confirm right now but it should help us move forward with these updates

**Comment by sjcobb** [comment]
Would love to get ECharts upgraded too in #1332 but running into issues with Jest

File: ui/storybook/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "storybook:build": "npm run storybook:clean && storybook build --config-dir src/config",
     "storybook:serve": "npx http-server --port=6006 ./storybook-static",
     "storybook:test": "test-storybook --config-dir src/config --stories-json",
-    "storybook:serve-and-test": "concurrently -k -s first -n \"SB,TEST\" \"npm run storybook:serve -- --silent\" \"npx wait-on tcp:6006 && npm run storybook:test\"",
+    "storybook:serve-and-test": "concurrently -k -s first -n \"SB,TEST\" \"npm run storybook:serve -- --silent\" \"npx wait-on tcp:127.0.0.1:6006 && npm run storybook:test\"",
```
Comment (sjcobb) [inline ui/storybook/package.json:12]: This change ended up fixing the hanging Storybook tests in CI

Comment (Gladorme) [reply ui/storybook/package.json:12]: Nice! 🤩 I can do some manual testing when I'm back on Monday 👍

**Review by Gladorme — APPROVED** [review]
Found only one issue when doing some tests 😄

File: ui/app/src/components/dialogs/RenameDashboardDialog.tsx [hunk 2]
```diff
@@ -37,22 +37,22 @@ export const RenameDashboardDialog = (props: RenameDashboardDialogProps) => {
   const { successSnackbar, exceptionSnackbar } = useSnackbar();
   const updateDashboardMutation = useUpdateDashboardMutation();
 
-  const [name, setName] = useState<string>();
-  const [error, setError] = useState<string>();
+  const [name, setName] = useState<string>('');
```
[the remark this answers is not in the document]
Comment (Gladorme) [reply ui/app/src/components/dialogs/RenameDashboardDialog.tsx:40]: I confirm the issue is related to my changes. `defaultValue` is probably working fine if `value` is `undefined`, but not when it's a empty string 😅

[the remark this answers is not in the document]
Comment (sjcobb) [reply ui/app/src/components/dialogs/RenameDashboardDialog.tsx:40]: Thanks! Can you push that change up to my `sjcobb/Gladorme-deps-update` and merge?

[the remark this answers is not in the document]
Comment (Gladorme) [reply ui/app/src/components/dialogs/RenameDashboardDialog.tsx:40]: Sure 👍

**Pull request merged**